### ISSUE TITLE
Use Harmony to apply volume/mute

### ISSUE THREAD

**Issue by 6uhrmittag** [StarriMods/BackgroundMusicSoftener#1]
Applying custom volume and mute is currently done in the `OnLateUpdate()` and `OnSceneWasLoaded()` Callback because it's the simplest method to change game behaviour(and explained in the [MelonLoader Quickstart](https://melonwiki.xyz/#/modders/quickstart) 😬 

However, it looks like Using [Harmony ](https://melonwiki.xyz/#/modders/patching) to patch the functions that get called directly is the go-to method for doing that.

The Background Music is controlled in different ways:

- in main menu:
    - Game Object name: `BackgroundAudioSource`
    - Assembly: `UnityEngine.AudioSource`
    - parameter for volume: `AudioSource.volume` (float)
    - parameter for mute: `AudioSource.mute` (bool)
    - Clip playing: `Starri_BGM_Loop_conv`
- in summary view after run:
    - Game Object name: `SummaryView(Clone)`
    - Assembly: `UnityEngine.AudioSource`
    - parameter for volume: `AudioSource.volume` (float)
    - parameter for mute: `AudioSource.mute` (bool)
    - Clip playing: `Starri_BGM_Loop_conv`

It seems that it rather sets the volume to 0 than actually mutes it in views like the Song Selection.
